### PR TITLE
fix(Tag):   The ellipsis with icons should work

### PR DESCRIPTION
**Pull request by linhf123**
### 📦 Modified package

- [x] @oceanbase/design
- [ ] @oceanbase/ui
- [ ] @oceanbase/icons
- [ ] @oceanbase/charts
- [ ] @oceanbase/util
- [ ] @oceanbase/codemod
- [ ] Other (about what?)

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

1. Icon with ellipsis is abnormal for tag

| Before | After |
| :--- | :--- |
| image|image|

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Typography uses custom Icon in the case of ellipsis        |
| 🇨🇳 Chinese |    Typography 省略情况下使用自定义 Icon       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Tests is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: packages/design/src/tag/index.tsx [hunk 1]
```diff
@@ -35,18 +35,28 @@ const Tag = React.forwardRef<HTMLSpanElement, TagProps>(
...
+        ref={ref}
+        prefixCls={customizePrefixCls}
+        className={tagCls}
+        icon={ellipsisConfig ? null : icon}
+        {...restProps}
+      >
         {ellipsisConfig ? (
-          <Typography.Text ellipsis={ellipsisConfig}>{children}</Typography.Text>
+          <Typography.Text ellipsis={ellipsisConfig}>
+            {realIcon}
```
Comment (dengfuping) [inline packages/design/src/tag/index.tsx:57]: icon 在 Tag 和 Typography 内都有设置，是特意的吗？原因是什么呢？

File: packages/design/src/tag/index.tsx [hunk 2]
```diff
@@ -35,18 +35,28 @@ const Tag = React.forwardRef<HTMLSpanElement, TagProps>(
...
     const ellipsisConfig = getEllipsisConfig(ellipsis, children);
-
     const tagCls = classNames(
       {
         [`${prefixCls}-ellipsis`]: !!ellipsisConfig,
       },
       className
     );
 
+    const realIcon = icon ? <span style={{ marginRight: 7 }}>{icon}</span> : null;
```
Comment (dengfuping) [inline packages/design/src/tag/index.tsx:45]: antd 7px 的计算逻辑，可以参考下，尽量不写死

https://github.com/ant-design/ant-design/blob/master/components/tag/style/index.ts#L35

**Review by dengfuping — CHANGES_REQUESTED** [review]

File: packages/design/src/tag/index.tsx (hunk 1 above)
Comment (linhf123) [reply packages/design/src/tag/index.tsx:57]: 主要是为了使用 Typography 的 ellipsis，Typography.Text 内部会判断是否超出宽度来展示 tooltip，看了下对应的实现逻辑，自己来实现溢出之后展示 Tooltip 有点麻烦，所以将 Icon 包进去展示进行绕过

File: packages/design/src/tag/index.tsx (hunk 2 above)
Comment (linhf123) [reply packages/design/src/tag/index.tsx:45]: ok，这里我改一下

**Comment by dengfuping** [comment]
CI 挂了，需要重新拉取下 master 代码

File: packages/design/src/tag/index.tsx [hunk 3]
```diff
@@ -35,18 +35,29 @@ const Tag = React.forwardRef<HTMLSpanElement, TagProps>(
...
+        prefixCls={customizePrefixCls}
+        className={tagCls}
+        icon={ellipsisConfig ? null : icon}
+        {...restProps}
+      >
         {ellipsisConfig ? (
-          <Typography.Text ellipsis={ellipsisConfig}>{children}</Typography.Text>
+          <Typography.Text ellipsis={ellipsisConfig}>
+            {/* Typography.Text 存在 ellipsis 配置时 ，将 icon 放在 Typography.Text 内部，避免溢出时与 icon 发生样式冲突。这里保留 Typography.Text 主要为了使用 Typography.Text 的判断内容溢出展示 Tooltip 的能力，自定义实现成本过大 */}
+            {realIcon}
```
Comment (dengfuping) [inline packages/design/src/tag/index.tsx:58]: 这里应该是 `{ ellipsisConfig ? realIcon : null }` 吧，即有 ellipsis 配置时，将 icon 挪到 Typography 内部

File: packages/design/src/tag/index.tsx [hunk 4]
```diff
@@ -35,18 +35,29 @@ const Tag = React.forwardRef<HTMLSpanElement, TagProps>(
...
     const ellipsisConfig = getEllipsisConfig(ellipsis, children);
-
     const tagCls = classNames(
       {
         [`${prefixCls}-ellipsis`]: !!ellipsisConfig,
       },
       className
     );
 
+    const realIcon = icon ? <span className="inner-icon">{icon}</span> : null;
```
Comment (dengfuping) [inline packages/design/src/tag/index.tsx:45]: 这里需要改为 Tag icon 的类名

**Review by dengfuping — CHANGES_REQUESTED** [review]

File: packages/design/src/tag/index.tsx (hunk 3 above)
Comment (linhf123) [reply packages/design/src/tag/index.tsx:58]: 这个 ellipsisConfig 的配置，外层已经包了
image

File: packages/design/src/tag/index.tsx (hunk 4 above)
Comment (linhf123) [reply packages/design/src/tag/index.tsx]: 已修改

**Pull request merged**